packages/lit-dev-content/site/blog/2022-07-19-lit-for-polymer-users.md

```diff
@@ -0,0 +1,935 @@
+
+<dl class="params">
+<dt class="paramName"><code>type</code></dt>
+<dd class="paramDetails">
+
+Lit's `type` option serves the same purpose.
+
+</dd>
+
+<dt class="paramName"><code>value</code></dt>
+<dd class="paramDetails">
+
+Lit doesn't support setting a property's value like this. Instead, set a default value in the constructor. If using the `@property` decorator, you can also use a class field initializer.
+
+</dd>
+
+<dt class="paramName"><code>reflectToAttribute</code></dt>
+<dd class="paramDetails">
+
+In Lit, this option has been shortened to `reflect`.
+
+</dd>
+
+<dt class="paramName"><code>readOnly</code></dt>
+<dd class="paramDetails">
+
+Lit doesn't include any built-in support for read-only reactive properties. If you need to make a calculated property part of a component's public API, you can add a getter with no setter.
+</dd>
+
+
+<dt class="paramName"><code>notify</code></dt>
+<dd class="paramDetails">
+
+Not supported in Lit. The `notify` option generates an event when the property changes, and is used to support two-way data binding. This option is problematic because the event is fired any time the property changes, even if it's changed programatically. Lit components can use the native web APIs (such as `dispatchEvent`) to fire events in response to user input or when an internal state changes (for example, when an animation completes).
+
+</dd>
+
+<dt class="paramName"><code>computed</code></dt>
+<dd class="paramDetails">
+
+Declarative computed properties are not supported in Lit. See computed properties for alternatives.
+
+</dd>
+
+<dt class="paramName"><code>observer</code></dt>
+<dd class="paramDetails">
+
+Lit doesn't support observers directly. Instead, it supplies a number of override points in the lifecycle where you can take action based on any properties that have changed.'
+</dd>
+</dl>
+
+#### Static getter versus static class field
+
+Note that the Polymer example uses a getter—`static get properties()`—where the Lit JavaScript example uses a class field—`static properties`. These two forms work the same way. When Polymer 3 was published, support for class fields was not widespread, so Polymer example code uses the getter.
+
+If you're adding Lit components to an existing Polymer application, your toolchain may not support the class field format. In which case, you can use the static getter, instead.
+
+```js
+static get properties() {`
+  return {
+    user: {},
+    count: {
+      type: Number
+    }
+  }
+}
+```
+
+
+#### Array & object type properties
+
+Like Polymer, Lit does dirty checking when properties change to avoid performing unnecessary work. This can lead to issues if you have a property that holds an object or array. If you mutate the object or array, Lit won't detect a change.
+
+In most cases, the best way to avoid these issues is to use immutable data patterns, so that you always assign a new object or array value instead of mutating an existing object or array.
```

Comment:
The rules for Polymer and Lit are the same if you don't include `this.set` so I think that needs to be mentioned here.

Reply:
Mention `this.set`? Or mention that the rules are the same?

Reply:
Mentioned rules are basically the same, added a paragraph mentioning `this.set` and friends (which will probably need to be replaced with immutable patterns when moving to Lit).